SU2_CFD/src/solvers/CSpeciesSolver.cpp

```diff
@@ -168,6 +178,109 @@ CSpeciesSolver::CSpeciesSolver(CGeometry* geometry, CConfig* config, unsigned sh
   SolverName = "SPECIES";
 }
 
+void CSpeciesSolver::BC_Fluid_Interface(CGeometry *geometry, CSolver **solver_container, CNumerics *conv_numerics,
+                                     CNumerics *visc_numerics, CConfig *config) {
+  const auto nPrimVar = solver_container[FLOW_SOL]->GetnPrimVar();
+  su2double *PrimVar_j = new su2double[nPrimVar];
+  su2double solution_j[MAXNVAR] = {0.0};
+
+  for (auto iMarker = 0u; iMarker < config->GetnMarker_All(); iMarker++) {
+
+    if (config->GetMarker_All_KindBC(iMarker) != FLUID_INTERFACE) continue;
+
+    SU2_OMP_FOR_STAT(OMP_MIN_SIZE)
+    for (auto iVertex = 0u; iVertex < geometry->nVertex[iMarker]; iVertex++) {
+
+      const auto iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
+
+      if (!geometry->nodes->GetDomain(iPoint)) continue;
+
+      const auto Point_Normal = geometry->vertex[iMarker][iVertex]->GetNormal_Neighbor();
+      const auto nDonorVertex = GetnSlidingStates(iMarker,iVertex);
+
+      su2double Normal[MAXNDIM] = {0.0};
+      for (auto iDim = 0u; iDim < nDim; iDim++)
+        Normal[iDim] = -geometry->vertex[iMarker][iVertex]->GetNormal()[iDim];
+
+      su2double* PrimVar_i = solver_container[FLOW_SOL]->GetNodes()->GetPrimitive(iPoint);
```

Comment:
Also looks copy pasted from the turbulence solver

Reply:
thanks! I've moved this up to scalarsolver

Reply:
> Also looks copy pasted from the turbulence solver
indeed I copied it from the turbulence solver :)